server/auth/types/openid/routes.ts

```diff
@@ -336,8 +336,10 @@ export class OpenIdAuthRoutes {
                 }
                 let params = new URLSearchParams(window.location.search);
                 let nextUrl = params.get("nextUrl");
-                finalUrl = "login?nextUrl=" + encodeURIComponent(nextUrl);
-                finalUrl += "&redirectHash=" + encodeURIComponent(redirectHash);
+                finalUrl = "login?redirectHash=" + encodeURIComponent(redirectHash);
+                if (!!nextUrl) {
```

Comment:
Yes, it is related. The [PR](https://github.com/opensearch-project/security-dashboards-plugin/pull/1563) to implement nextUrl for OpenID auth introduced a regression for instances of OSD using a serverBasePath. The reason was that on login, it would always set nextUrl to `/` instead of `/{serverBasePath}` and after successful login it would redirect back to the wrong url.   

OpenID auth uses this method ([`getNextUrl`](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/server/auth/types/openid/helper.ts#L75-L81)) to get the URL to redirect to after successful login and it was never reaching the `or` block on this [line](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/server/auth/types/openid/helper.ts#L75-L81).

This PR makes it so that nextUrl is only added on login if the user had a previously expired session or explicitly logs out. If the user is loading OSD for the first time (like in an incognito window), there will be no `nextUrl`. If the user is freshly loading OSD, then this will route them to their serverBasePath/defaultRoute on successful login.

Reply:
Thanks for the extra context @cwperks ! Yes I think I understand it now.